marker_rustc_driver/src/main.rs

```diff
@@ -75,6 +76,35 @@ impl rustc_driver::Callbacks for MarkerCallback {
             register_tracked_files(sess, &lint_crates);
         }));
 
+        // Make it possible to use `#[allow(marker::{lint_name})]` without
+        // having to add `#![feature(register_tool)]` and `#![register_tool(marker)]`.
+        config.override_queries = Some(|_sess, providers, _extern_providers| {
```

Comment:
I didn't even know that it was possible to override quarries o.O. 

In red pen, I found another hack, that adds the attributes via console arguments. Have you considered that as well?

https://github.com/estebank/redpen/blob/b4008d2092ff4fdbb3f735d1892a88a63c956d97/commands/wrapper.rs#L39-L40

I'm not sure if that's more or less sustainable than this option.

The best would be to have an official interface in rustc for this.

Reply:
Yeah, that may be a good idea. I wonder if it could be configured at workspace level, if that's a cli option it could be part of rustflags, I'll think what it may give us

Reply:
I though that we could have a thin rustc wrapper script e.g. `cargo marker rustc` that users could put into `RUSTC_WRAPPER` to avoid writing `cfg_attr` and any other ugly boilerplate. That wrapper would add the `-Zcrate-attr=feature(register_tool)`, and `-Zcrate-attr=register_tool(marker)` automatically to any `rustc` invocation. But.. there is a problem with this approach:

`-Zcrate-attr` is an unstable option, so users can't have it on stable toolchain. We could potentially add `RUSTC_BOOTSTRAP=1` env variable to workaround that, but that env variable affects the entire `rustc` CLI invocation, meaning users can now use any unstable options, because our wrapper script adds that env variable for them. That's quite ugly..

So for having this inside of the marker rustc driver only and requiring `cfg_attr` is the best option for now.